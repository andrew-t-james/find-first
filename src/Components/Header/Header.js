import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

export const Header = ({ user }) => {

  return (
    <div className="header">
      <h1>header</h1>
      {user.name &&
         <div className="user-info">
           <h2 className="user-info__heading">{user.name}</h2>
           <img className="user-info__image" src={user.image} alt={user.name}/>
         </div>
      }
    </div>
  );
};

export const mapStateToProps = state => ({
  user: state.user
});

export default connect(mapStateToProps)(Header);
