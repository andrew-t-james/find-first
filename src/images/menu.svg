<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px"
fill="#EAECEE" viewBox="0 0 396.9 496.125" enable-background="new 0 0 396.9 396.9" xml:space="preserve"><g><rect x="85" y="258.4" width="226.8" height="14.6"/></g><g><rect x="85" y="124" width="226.8" height="14.6"/></g><g><rect x="85" y="191.2" width="138.7" height="14.6"/></g></svg>