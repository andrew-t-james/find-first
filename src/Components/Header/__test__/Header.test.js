import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from '../Header';
import { toggleMenu } from '../../../Actions/menu';

describe('<Header />', () => {
  let wrapper;
  const mockUser = {
    name: 'Joben',
    image: 'some-url-here'
  };
  const mockState = {
    savedJobs: [1, 2]
  };
  const mockSliderAction = true;

  beforeEach(() => wrapper = shallow(<Header
    toggleMenu={mockSliderAction}
    user={mockUser}
    recentJobs={mockState}
  />
  ));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    const expected = {
      user: mockUser
    };
    const mockState = {
      user: mockUser
    };
    const result = mapStateToProps(mockState);
    expect(result).toEqual(expected);
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch when toggleMenu is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleMenu();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.toggleMenu();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });

});
