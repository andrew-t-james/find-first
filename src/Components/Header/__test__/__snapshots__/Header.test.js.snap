// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<section
  className="header"
>
  <div
    className="header-toggle wrapper-menu"
  >
    <div
      className="line-menu half start"
    />
    <div
      className="line-menu"
    />
    <div
      className="line-menu half end"
    />
  </div>
  <div
    className="header__logo"
  >
    <h1
      className="logo__forward"
    >
      F
    </h1>
    <h1
      className="logo__backward"
    >
      F
    </h1>
  </div>
  <div
    className="user-info"
  >
    <h2
      className="user-info__heading"
    >
      Joben
    </h2>
    <img
      alt="Joben"
      className="user-info__image"
      src="some-url-here"
    />
  </div>
</section>
`;
